front/pages/w/[wId]/subscribe.tsx

```diff
@@ -0,0 +1,126 @@
+import { BarHeader, Button, Page } from "@dust-tt/sparkle";
+import type { SubscriptionType } from "@dust-tt/types";
+import type { WorkspaceType } from "@dust-tt/types";
+import { CreditCardIcon } from "@heroicons/react/20/solid";
+import type { InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+import { useContext } from "react";
+
+import { AssistantSidebarMenu } from "@app/components/assistant/conversation/SidebarMenu";
+import { ProPriceTable } from "@app/components/PlansTables";
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { useSubmitFunction } from "@app/lib/client/utils";
+import { withDefaultUserAuthRequirements } from "@app/lib/iam/session";
+
+const { GA_TRACKING_ID = "" } = process.env;
+
+export const getServerSideProps = withDefaultUserAuthRequirements<{
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  gaTrackingId: string;
+}>(async (context, auth) => {
+  const owner = auth.workspace();
+  const subscription = auth.subscription();
+  if (!owner || !auth.isUser() || !subscription) {
```

Comment:
Technically you'll have a subscription (the default one that will be with code `FREE_NO_PLAN`) but we really don't need to check for this page which is meant to create it :)

Reply:
Just for the record, this PR was not ready for review when this comment was posted, a lot of code has been removed since then (including the <AppLayout/>), which implies removing the subscription check.